packages/core/echo/indexing/src/index-schema.ts

```diff
@@ -1,113 +1,78 @@
 
-  private _orama?: OramaSchemaType = undefined;
-
-  override async _open() {
-    this._orama = await orama.create({
-      schema: {
-        system: {
-          type: { itemId: 'string' },
-        },
-      },
-    });
-  }
-
-  override async _close() {}
+  /**
+   * Map `typename` -> Set `index id`.
+   */
+  private readonly _index = new Map<string | null, Set<ObjectPointerEncoded>>();
 
   get identifier() {
     return this._identifier;
   }
 
   @trace.span({ showInBrowserTimeline: true })
   async update(id: string, object: Partial<ObjectStructure>) {
-    invariant(this._orama, 'Index is not initialized');
-    const entry = await orama.getByID(this._orama, id);
-    if (entry && entry.system.type?.itemId === object.system?.type?.itemId) {
+    if (this._index.get(getTypeFromObject(object))?.has(id)) {
       return false;
     }
-    await orama.update<any>(this._orama, id, { ...object, id });
+    defaultMap(this._index, getTypeFromObject(object), new Set()).add(id);
     return true;
   }
 
   async remove(id: string) {
-    invariant(this._orama, 'Index is not initialized');
-    await orama.remove(this._orama, id);
+    for (const [_, ids] of this._index.entries()) {
+      if (ids.has(id)) {
+        ids.delete(id);
+        return;
+      }
+    }
   }
 
-  // TODO(mykola): Fix Filter type with new Reactive API.
   @trace.span({ showInBrowserTimeline: true })
   async find(filter: Filter) {
-    invariant(this._orama, 'Index is not initialized');
-    let results: orama.Results<Partial<ObjectStructure>>;
-    if (!filter.type) {
-      results = await orama.search(this._orama, {
-        term: '',
-        exact: true,
-        threshold: 1,
-        limit: ORAMA_LIMIT,
-      });
-    } else {
-      results = await orama.search<OramaSchemaType, Partial<ObjectStructure>>(this._orama, {
-        term: filter.type.itemId,
-        exact: true,
-        threshold: 0,
-        limit: ORAMA_LIMIT,
-      });
-    }
-
-    return results.hits.map((hit) => ({ id: hit.id, rank: hit.score }));
+    return Array.from(this._index.get(filter.type?.itemId ?? null) ?? []).map((id) => ({ id, rank: 0 }));
   }
 
   @trace.span({ showInBrowserTimeline: true })
   async serialize(): Promise<string> {
-    invariant(this._orama, 'Index is not initialized');
-    return JSON.stringify(await orama.save(this._orama), null, 2);
+    return JSON.stringify({
+      index: Array.from(this._index.entries()).map(([type, ids]) => ({
+        type,
+        ids: Array.from(ids),
+      })),
+    });
   }
 
   @trace.span({ showInBrowserTimeline: true })
   static async load({ serialized, identifier }: LoadParams): Promise<IndexSchema> {
-    const deserialized = JSON.parse(serialized);
-
     const index = new IndexSchema();
-    await index.open();
-    invariant(index._orama, 'Index is not initialized');
+    const serializedIndex: Array<{ type: string | null; ids: string[] }> = JSON.parse(serialized).index;
```

Comment:
stick with `T[]` syntax for arrays 

```suggestion
    const serializedIndex: { type: string | null; ids: string[] }[] = JSON.parse(serialized).index;
```